pkg/instance/service.go

```diff
@@ -230,34 +221,23 @@ func (s service) unlink(id uint) error {
 	return s.instanceRepository.Unlink(instance)
 }
 
-func (s service) Create(instance *model.Instance) (*model.Instance, error) {
-	err := s.instanceRepository.Create(instance)
-	if err != nil {
-		return nil, err
-	}
-
-	instanceWithParameters, err := s.instanceRepository.FindWithParametersById(instance.ID)
+func (s service) Save(instance *model.Instance) (*model.Instance, error) {
+	enrichParameters(instance)
```

Comment:
If I read correctly you are calling `enrichParameters` in the repo.Save as well 😅  In that case, I don't think its needed.